libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -843,6 +848,8 @@ unsigned CompilerUtils::prepareMemoryStore(Type const& _type, bool _padToWordBou
 	else
 	{
 		solAssert(numBytes <= 32, "Memory store of more than 32 bytes requested.");
+		if (_type.category() == Type::Category::Bool)
```

Comment:
I need to clean all types here.  `ExpressionCompiler::appendStateVariableAccessor` calls `storeInMemory` calls `prepareMemoryStore` without cleaning the topmost stack element.